packages/did/src/didAuthorizeExtrinsics/didAuthorizeExtrinsics.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2018-2022, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import type { Extrinsic } from '@polkadot/types/interfaces'
+import type { SubmittableExtrinsicFunction } from '@polkadot/api/types'
+
+import { BN } from '@polkadot/util'
+import type {
+  DidDetails,
+  DidVerificationKey,
+  IIdentity,
+  SignCallback,
+  SubmittableExtrinsic,
+  VerificationKeyRelationship,
+} from '@kiltprotocol/types'
+import { SDKErrors } from '@kiltprotocol/utils'
+
+import {
+  authorizeExtrinsic,
+  getNextNonce,
+} from '../DidDetails/FullDidDetails.js'
+import {
+  getKeyRelationshipForExtrinsic,
+  increaseNonce,
+} from '../DidDetails/FullDidDetails.utils.js'
+
+import { generateDidAuthenticatedTx } from '../Did.chain.js'
+import { getSigningAlgorithmForVerificationKeyType } from '../Did.utils.js'
+
+/**
+ * Type of a callback used to select one of the key candidates for a DID to sign a given batch of extrinsics.
+ */
+type SelectKeyCallback = (
+  keys: DidVerificationKey[],
+  batch?: Extrinsic[]
```

Comment:
If it's an array, why do we make it optional an not just default to the empty one? Wouldn't it make it easier to deal with it?

Reply:
Nice catch! Looks like I wanted to use the same thing for authorizeExtrinsic and it failed, so instead of fixing the DidKeySelectionCallback I made the batch here optional. It doesn’t make sense like this I think. I’ll have to fix it properly.

Reply:
I’ve pushed a fix for that